collect_app/src/main/java/org/odk/collect/android/upload/UploadException.java

```diff
@@ -0,0 +1,33 @@
+/*
+ * Copyright (C) 2018 Nafundi
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except
+ * in compliance with the License. You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed under the License
+ * is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express
+ * or implied. See the License for the specific language governing permissions and limitations under
+ * the License.
+ */
+
+package org.odk.collect.android.upload;
+
+/**
+ * An exception that results in the cancellation of an instance upload, and the presentation of an
+ * error to the user
+ * */
+public class UploadException extends Exception {
```

Comment:
At the moment it looks like the result of `getDisplayMessage()` (if not also the `message` field itself) is a mixture of localizable and non-localizable strings, since sometimes a localized message is passed in and sometimes an exception.  Also, in the various places where an `UploadException` is constructed from another exception, sometimes it's done with `new UploadException(e)` and sometimes `new UploadException(e.getMessage())`.

I think it's okay to postpone addressing the localization issue for now, though it would be good to:

  - Document that this is an intentional but temporary state of affairs
  - Always construct with `new UploadException(e)` and not `new UploadException(e.getMessage())` so that we can preserve information we might use later to decide what to do about localization